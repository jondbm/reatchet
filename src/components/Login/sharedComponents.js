import styled from 'styled-components'

import Recaptcha from '../common/Recaptcha'
import theme from '../../theme'

export const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  font-family: Lato, sans-serif;
  max-width: 100%;
`

export const Title = styled.span`
  text-align: center;
  font-size: 20px;
  font-weight: bold;
  color: ${theme.colours.steel};
  margin-bottom: ${theme.spacing.sm};
  max-width: 100%;
`
export const Subtitle = styled.span`
  font-size: 16px;
  text-align: center;
  color: ${theme.colours.greyDark};
  margin-bottom: ${theme.spacing.sm};
  max-width: 100%;
`

export const StyledRecaptcha = styled(Recaptcha)`
  margin-top: ${theme.spacing.xs};
`
