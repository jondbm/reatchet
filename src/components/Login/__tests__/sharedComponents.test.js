import React from 'react'
import 'jest-styled-components'
import { shallow } from 'enzyme'

import { Wrapper, Title, Subtitle, StyledRecaptcha } from '../sharedComponents'

describe('Shared components', () => {
  describe('<Wrapper /> component', () => {
    let component

    beforeEach(() => {
      component = shallow(<Wrapper />)
    })

    it('should correctly render the component', () => {
      expect(component.exists()).toBe(true)
    })

    it('should match the snapshot', () => {
      expect(component).toMatchSnapshot()
    })
  })

  describe('<Title /> component', () => {
    let component

    beforeEach(() => {
      component = shallow(<Title />)
    })

    it('should correctly render the component', () => {
      expect(component.exists()).toBe(true)
    })

    it('should match the snapshot', () => {
      expect(component).toMatchSnapshot()
    })
  })

  describe('<Subtitle /> component', () => {
    let component

    beforeEach(() => {
      component = shallow(<Subtitle />)
    })

    it('should correctly render the component', () => {
      expect(component.exists()).toBe(true)
    })

    it('should match the snapshot', () => {
      expect(component).toMatchSnapshot()
    })
  })

  describe('<StyledRecaptcha /> component', () => {
    let component

    beforeEach(() => {
      component = shallow(<StyledRecaptcha />)
    })

    it('should correctly render the component', () => {
      expect(component.exists()).toBe(true)
    })

    it('should match the snapshot', () => {
      expect(component).toMatchSnapshot()
    })
  })
})
