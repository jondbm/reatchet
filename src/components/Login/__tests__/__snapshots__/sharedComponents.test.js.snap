// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shared components <StyledRecaptcha /> component should match the snapshot 1`] = `
.c0 {
  margin-top: 12px;
}

<Recaptcha
  className="c0"
/>
`;

exports[`Shared components <Subtitle /> component should match the snapshot 1`] = `
.c0 {
  font-size: 16px;
  text-align: center;
  color: #999999;
  margin-bottom: 24px;
  max-width: 100%;
}

<span
  className="c0"
/>
`;

exports[`Shared components <Title /> component should match the snapshot 1`] = `
.c0 {
  text-align: center;
  font-size: 20px;
  font-weight: bold;
  color: #16222a;
  margin-bottom: 24px;
  max-width: 100%;
}

<span
  className="c0"
/>
`;

exports[`Shared components <Wrapper /> component should match the snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-family: Lato,sans-serif;
  max-width: 100%;
}

<div
  className="c0"
/>
`;
